File: 30-days-of-code/java/day-06-lets-review.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static void main(String[] args) {
        // Begin Solution
        //---------------------------------------------------------------------------------------
        int nb_tests;
        Scanner scan = new Scanner(System.in);
        nb_tests = scan.nextInt();
        scan.nextLine();
        for (int i=0; i<nb_tests; i++) {
            String my_string;
            String left_side = "";
            String right_side = "";
            my_string = scan.nextLine();
            char[] myCharArray = my_string.toCharArray();
            for (int j=0; j<my_string.length();j++){
                if (j % 2 == 0) {
                    left_side += myCharArray[j];
                } else {
                    right_side += myCharArray[j];
                };
            };
            System.out.println(String.format("%s%s%s", left_side, " ", right_side));
        }
        //---------------------------------------------------------------------------------------
        // End Solution
    }
}

File: 30-days-of-code/cpp/day-06-lets-review.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    // Begin Solution
    //---------------------------------------------------------------------------------------
    int nb_tests;
    cin >> nb_tests;
    for (int i=0; i<nb_tests; i++) {
        string my_string;
        string left_side;
        string right_side;
        cin >> my_string;
        for (int j=0; j<my_string.length();j++) {
            if (j%2==0) {
                left_side += my_string[j];
            } else {
                right_side += my_string[j];
            };
        };
        cout << left_side << " " << right_side << endl;
    };
    //---------------------------------------------------------------------------------------
    // End Solution

    return 0;
}
